### PR TITLE
Add QUIC Version 1 Gap Analysis, replacing conclusions section.

### LINKED ISSUE #63: Conclusion
There is a Conclusion section we need either to fill or to remove.

### PR DESCRIPTION
**Pull request by SpencerDawkins**
This is where we are saying, in a small number of words, what we hope the IETF will provide in QUIC extensions so that we can use those extensions in the next version of ATSSS.

Please review this carefully - it needs to be short, but complete.

### CODE REVIEW

File: draft-quic-atsss-reqs.mkd [hunk 1]
```diff
@@ -706,10 +708,17 @@ Volume-based Quota
 
 Wireless LAN (WLAN)
 
-# Conclusion {#conclusion}
+# QUIC Version 1 Gap Analysis for ATSSS Version 2 {#gap-analysis}
+
+These are additional QUIC protocol capabilities that would be useful for ATSSS Phase 2, as described in {{sec-ver2}}, along with our understanding of their maturity level.
+
+1. ATSSS Version 2 will need transport services for Ethernet frames and IP packets, with the intention of supporting TCP, UDP, and UDP-encapsulated transport protocols such as QUIC and RTP. QUIC Version 1 provides transport services for HTTP/3 but does not yet provide support for tunnelling Ethernet or IP directly over QUIC, as envisioned in ATSSS Version 2. QUIC Version 1. The QUIC working group is not chartered to provide this capability, although proposals exist.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
1. ATSSS Version 2 will need transport services for Ethernet frames and IP packets, with the intention of supporting any traffic (including, TCP, UDP, and UDP-encapsulated transport protocols such as QUIC). QUIC Version 1 provides transport services for HTTP/3 but does not yet provide support for tunnelling Ethernet or IP directly over QUIC.  The QUIC working group is not chartered to provide this capability, although proposals exist.
```

I would put this one as the last bullet item.

File: draft-quic-atsss-reqs.mkd [hunk 2]
```diff
@@ -706,10 +708,17 @@ Volume-based Quota
...
+
+1. QUIC Version 1 does not yet provide a multipath transport capability similar to Multipath TCP, as used in ATSSS Phase 1. The QUIC working group is chartered to produce a multipath extension document by December 2021. Proposals exist, although the working group has not yet adopted a draft as the basis for its work.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
1. QUIC Version 1 does not yet provide a multipath transport capability similar to Multipath TCP, as used in ATSSSv1. The QUIC working group is chartered to produce a multipath extension document by December 2021. Proposals exist, although the working group has not yet adopted a draft as the basis for its work. 
```

File: draft-quic-atsss-reqs.mkd [hunk 3]
```diff
@@ -706,10 +708,17 @@ Volume-based Quota
...
+
+1. ATSSS Version 2 will need transport services for Ethernet frames and IP packets, with the intention of supporting TCP, UDP, and UDP-encapsulated transport protocols such as QUIC and RTP. QUIC Version 1 provides transport services for HTTP/3 but does not yet provide support for tunnelling Ethernet or IP directly over QUIC, as envisioned in ATSSS Version 2. QUIC Version 1. The QUIC working group is not chartered to provide this capability, although proposals exist.
+
+1. QUIC Version 1 does not yet provide a multipath transport capability similar to Multipath TCP, as used in ATSSS Phase 1. The QUIC working group is chartered to produce a multipath extension document by December 2021. Proposals exist, although the working group has not yet adopted a draft as the basis for its work. 
+
+1. For Multipath QUIC, we envision the ability to select paths (steering), detect path failures and reroute traffic (switching), and forward packets on multiple active paths simultaneously (splitting), based on external policies, including active-standby, smallest delay, weighted load-balancing, and path selection based on assigned priorities.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]: I would put this one before the previous bullet.

File: draft-quic-atsss-reqs.mkd (hunk 1 above)
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
1. ATSSS Version 2 will need transport services for Ethernet frames and IP packets, with the intention of supporting TCP, UDP, and UDP-encapsulated transport protocols such as QUIC and RTP. QUIC Version 1 provides transport services for HTTP/3 but does not yet provide support for tunnelling Ethernet or IP directly over QUIC, as envisioned in ATSSS Version 2. The QUIC working group is not chartered to provide this capability, although proposals exist.
```

File: draft-quic-atsss-reqs.mkd (hunk 2 above)
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]: Is that 1. part of an enumeration ?

**Comment by markusa** [comment]
Hi Spencer,

should we further mention in the conclusion that

- MP-QUIC has to considere demands from a potential re-assembly function, such as sequencing?
- CC in CC effects requires further investigation?
- multiple encrypted header might be optimized at 3GPP or IETF?

File: draft-quic-atsss-reqs.mkd [hunk 4]
```diff
@@ -706,10 +708,17 @@ Volume-based Quota
...
+
+1. QUIC Version 1 does not yet provide a multipath transport capability similar to Multipath TCP, as used in ATSSS Phase 1. The QUIC working group is chartered to produce a multipath extension document by December 2021. Proposals exist, although the working group has not yet adopted a draft as the basis for its work. 
+
+1. For Multipath QUIC, we envision the ability to select paths (steering), detect path failures and reroute traffic (switching), and forward packets on multiple active paths simultaneously (splitting), based on external policies, including active-standby, smallest delay, weighted load-balancing, and path selection based on assigned priorities. 
 
-Based on the two above questions, it should be clear that MPQUIC could
-cover both cases.
+1. QUIC Version 1 does not yet provide support for unreliable datagrams, needed for the additional user-plane protocols as envisioned in ATSSS Version 2, although the QUIC working group is chartered to produce this capability by March 2021, and has adopted a proposal as a working group draft.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
1. The base QUIC Version 1 does not provide support for unreliable datagrams, needed for the additional user-plane protocols as envisioned in ATSSS Version 2. The QUIC working group is chartered to produce this capability by March 2021, and has adopted a proposal as a working group document.
```

File: draft-quic-atsss-reqs.mkd (hunk 1 above)
Comment (boucadair) [reply draft-quic-atsss-reqs.mkd]: Also, I would position this bullet as the the last one in the list.

File: draft-quic-atsss-reqs.mkd [hunk 5]
```diff
@@ -706,10 +708,17 @@ Volume-based Quota
 
 Wireless LAN (WLAN)
 
-# Conclusion {#conclusion}
+# QUIC Version 1 Gap Analysis for ATSSS Version 2 {#gap-analysis}
+
+These are additional QUIC protocol capabilities that would be useful for ATSSS Phase 2, as described in {{sec-ver2}}, along with our understanding of their maturity level.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd:713]:
```suggestion
These are additional QUIC protocol capabilities that would be useful for ATSSS Version 2, as described in {{sec-ver2}}, along with our understanding of their maturity level.
```

**Review by boucadair — CHANGES_REQUESTED** [review]

**Comment by SpencerDawkins** [comment]
I'll make these changes. I have questions about whether reversing the order of 

1. QUIC Version 1 does not yet provide a multipath transport capability similar to Multipath TCP, as used in ATSSS Phase 1. The QUIC working group is chartered to produce a multipath extension document by December 2021. Proposals exist, although the working group has not yet adopted a draft as the basis for its work. 

1. For Multipath QUIC, we envision the ability to select paths (steering), detect path failures and reroute traffic (switching), and forward packets on multiple active paths simultaneously (splitting), based on external policies, including active-standby, smallest delay, weighted load-balancing, and path selection based on assigned priorities. 

makes sense, but I think that dropping "For Multipath QUIC" out of the second bullet removes my questions.

File: draft-quic-atsss-reqs.mkd (hunk 1 above)
Comment (SpencerDawkins) [reply draft-quic-atsss-reqs.mkd]: Done.

File: draft-quic-atsss-reqs.mkd (hunk 2 above)
Comment (SpencerDawkins) [reply draft-quic-atsss-reqs.mkd]: No, just part of Spencer doing a PR too late at night. Deleted :-)

**Comment by boucadair** [comment]
Hi Spencer,

The reasoning for reverting the order is that the second bullet defines the needs that are satisfied with multipath (first bullet). 

"For Multipath QUIC, " should be removed for the second bullet, indeed.

Thank you.

File: draft-quic-atsss-reqs.mkd.bak [hunk 6]
```diff
@@ -0,0 +1,749 @@
+---
```
Comment (markusa) [inline draft-quic-atsss-reqs.mkd.bak]: This file should be deleted

Comment (SpencerDawkins) [reply draft-quic-atsss-reqs.mkd.bak]: You're right! My apologies.

File: draft-quic-atsss-reqs.mkd [hunk 7]
```diff
@@ -706,10 +708,23 @@ Volume-based Quota
 
 Wireless LAN (WLAN)
 
-# Conclusion {#conclusion}
+# QUIC Version 1 Gap Analysis for ATSSS Version 2 {#gap-analysis}
+
+These are additional QUIC protocol capabilities that would be useful for ATSSS Version 2, as described in {{sec-ver2}}, along with our understanding of their maturity level.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
These are additional QUIC protocol capabilities that would be useful for ATSSS Phase 2, as described in {{sec-ver2}}, along with our understanding of their maturity level.
```

to align with #94

File: draft-quic-atsss-reqs.mkd [hunk 8]
```diff
@@ -706,10 +708,23 @@ Volume-based Quota
...
+
+1. ATSSS Version 2 will need transport services for Ethernet frames and IP packets (with the intention of supporting TCP, UDP, and UDP-encapsulated transport protocols such as QUIC). QUIC Version 1 provides transport services for HTTP/3 but does not yet provide support for tunnelling Ethernet or IP directly over QUIC, as envisioned in ATSSS Version 2. The QUIC working group is not chartered to provide this capability, although proposals exist.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
- ATSSS Phase 2 will need transport services for Ethernet frames and IP packets (with the intention of supporting TCP, UDP, and UDP-encapsulated transport protocols such as QUIC). QUIC Version 1 provides transport services for HTTP/3 but does not yet provide support for tunnelling Ethernet or IP directly over QUIC, as envisioned in ATSSS Phase 2. The QUIC working group is not chartered to provide this capability, although proposals exist.
```

File: draft-quic-atsss-reqs.mkd [hunk 9]
```diff
@@ -706,10 +708,23 @@ Volume-based Quota
...
+
+1. We envision the ability to select paths (steering), detect path failures and reroute traffic (switching), and forward packets on multiple active paths simultaneously (splitting), based on external policies, including active-standby, smallest delay, weighted load-balancing, and path selection based on assigned priorities, for the full range of encapsulated protocols in ATSSS Version 2, similar to the abilities provided by Multipath TCP for ATSSS Version 1.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
- We envision the ability to select paths (steering), detect path failures and reroute traffic (switching), and forward packets on multiple active paths simultaneously (splitting), based on external policies, including active-standby, smallest delay, weighted load-balancing, and path selection based on assigned priorities, for the full range of encapsulated protocols in ATSSS Phase 2, similar to the abilities provided by Multipath TCP for ATSSS Phase 1. 
```

File: draft-quic-atsss-reqs.mkd [hunk 10]
```diff
@@ -706,10 +708,23 @@ Volume-based Quota
...
+
+1. We envision the ability to select paths (steering), detect path failures and reroute traffic (switching), and forward packets on multiple active paths simultaneously (splitting), based on external policies, including active-standby, smallest delay, weighted load-balancing, and path selection based on assigned priorities, for the full range of encapsulated protocols in ATSSS Version 2, similar to the abilities provided by Multipath TCP for ATSSS Version 1. 
+
+1. QUIC Version 1 does not provide a multipath transport capability similar to Multipath TCP, as used in ATSSS Version 1. The QUIC working group is chartered to produce a multipath extension document by December 2021. Proposals exist, although the working group has not yet adopted a draft as the basis for its work.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
- QUIC Version 1 does not provide a multipath transport capability similar to Multipath TCP, as used in ATSSS Phase 1. The QUIC working group is chartered to produce a multipath extension document by December 2021. Proposals exist, although the working group has not yet adopted a draft as the basis for its work. 
```

File: draft-quic-atsss-reqs.mkd [hunk 11]
```diff
@@ -706,10 +708,23 @@ Volume-based Quota
...
+
+1. We envision the ability to select paths (steering), detect path failures and reroute traffic (switching), and forward packets on multiple active paths simultaneously (splitting), based on external policies, including active-standby, smallest delay, weighted load-balancing, and path selection based on assigned priorities, for the full range of encapsulated protocols in ATSSS Version 2, similar to the abilities provided by Multipath TCP for ATSSS Version 1. 
+
+1. QUIC Version 1 does not provide a multipath transport capability similar to Multipath TCP, as used in ATSSS Version 1. The QUIC working group is chartered to produce a multipath extension document by December 2021. Proposals exist, although the working group has not yet adopted a draft as the basis for its work. 
+
+1. QUIC Version 1 does not yet provide support for unreliable datagrams, needed for the additional user-plane protocols as envisioned in ATSSS Version 2. The QUIC working group is chartered to produce this capability by March 2021, and has adopted a proposal as a working group draft.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
- QUIC Version 1 does not provide support for unreliable datagrams, needed for the additional user-plane protocols as envisioned in ATSSS Phase 2. The QUIC working group is chartered to produce this capability by March 2021, and has adopted a proposal as a working group draft.
```

File: draft-quic-atsss-reqs.mkd [hunk 12]
```diff
@@ -706,10 +708,23 @@ Volume-based Quota
...
+
+1. We recognize that Multipath QUIC may not provide reassembly or appropriate sequencing for the encapsulated protocols.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
- We recognize that Multipath QUIC may not provide reassembly or appropriate sequencing for the encapsulated protocols. 
```

File: draft-quic-atsss-reqs.mkd [hunk 13]
```diff
@@ -706,10 +708,23 @@ Volume-based Quota
...
+
+1. QUIC Version 1 does not provide a multipath transport capability similar to Multipath TCP, as used in ATSSS Version 1. The QUIC working group is chartered to produce a multipath extension document by December 2021. Proposals exist, although the working group has not yet adopted a draft as the basis for its work. 
+
+1. QUIC Version 1 does not yet provide support for unreliable datagrams, needed for the additional user-plane protocols as envisioned in ATSSS Version 2. The QUIC working group is chartered to produce this capability by March 2021, and has adopted a proposal as a working group draft.
+
+1. We recognize that Multipath QUIC may not provide reassembly or appropriate sequencing for the encapsulated protocols. 
+
+1. We recognize that nested congestion control mechanisms (such as QUIC encapsulated in QUIC) require further study.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
- We recognize that nested congestion control mechanisms (such as QUIC encapsulated in QUIC) have implications that require further study. 
```

File: draft-quic-atsss-reqs.mkd [hunk 14]
```diff
@@ -706,10 +708,23 @@ Volume-based Quota
...
+
+1. QUIC Version 1 does not yet provide support for unreliable datagrams, needed for the additional user-plane protocols as envisioned in ATSSS Version 2. The QUIC working group is chartered to produce this capability by March 2021, and has adopted a proposal as a working group draft.
+
+1. We recognize that Multipath QUIC may not provide reassembly or appropriate sequencing for the encapsulated protocols. 
+
+1. We recognize that nested congestion control mechanisms (such as QUIC encapsulated in QUIC) require further study. 
 
-Based on the two above questions, it should be clear that MPQUIC could
-cover both cases.
+1. We recognize that the complete ATSSS Version 2 protocol stack would include encrypted headers at multiple levels, and that if this is a problem for ATSSS, it is likely 3GPP's problem to solve.
```
Comment (boucadair) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
- We recognize that the complete ATSSS Phase 2 protocol stack would include encrypted headers at multiple levels, and that if this is a problem for ATSSS, it is likely 3GPP's problem to solve. Likewise, the implications of the various encapsulation overhead is to be further assessed within 3GPP.
```

**Review by boucadair — CHANGES_REQUESTED** [review]

File: draft-quic-atsss-reqs.mkd (hunk 5 above)
Comment (mpiraux) [reply draft-quic-atsss-reqs.mkd:713]: Not sure what append here.

**Review by boucadair — APPROVED** [review]

**Review by mpiraux — APPROVED** [review]

**Pull request merged**